pallets/rc-migrator/src/staking/message.rs

```diff
@@ -200,318 +129,629 @@ pub struct StakingValues<Balance> {
+		values.max_nominators_count.map(MaxNominatorsCount::<T>::put);
+		values.active_era.map(|active_era| {
+			let active_era: pallet_staking_async::ActiveEraInfo = active_era.into();
+			ActiveEra::<T>::put(&active_era);
+			CurrentEra::<T>::put(active_era.index);
+		});
+		values.force_era.map(|force_era| {
+			let force_era: pallet_staking_async::Forcing = force_era.into();
+			ForceEra::<T>::put(force_era);
+		});
+		values.max_staked_rewards.map(MaxStakedRewards::<T>::put);
+		values.slash_reward_fraction.map(SlashRewardFraction::<T>::put);
+		values.canceled_slash_payout.map(CanceledSlashPayout::<T>::put);
+		// CurrentPlannedSession is not migrated
+		values.chill_threshold.map(ChillThreshold::<T>::put);
+	}
+}
+
+pub type PortableStakingValues = StakingValues<u128>;
+
+#[derive(
+	PartialEq,
+	Eq,
+	Clone,
+	Encode,
+	Decode,
+	DecodeWithMemTracking,
+	RuntimeDebug,
+	TypeInfo,
+	MaxEncodedLen,
+)]
+pub struct PortableActiveEraInfo {
+	pub index: EraIndex,
+	pub start: Option<u64>,
+}
+
+impl IntoPortable for pallet_staking::ActiveEraInfo {
+	type Portable = PortableActiveEraInfo;
+
+	fn into_portable(self) -> Self::Portable {
+		PortableActiveEraInfo { index: self.index, start: self.start }
+	}
+}
+
+impl Into<pallet_staking_async::ActiveEraInfo> for PortableActiveEraInfo {
+	fn into(self) -> pallet_staking_async::ActiveEraInfo {
+		pallet_staking_async::ActiveEraInfo { index: self.index, start: self.start }
+	}
+}
+
+#[derive(
+	PartialEq,
+	Eq,
+	Clone,
+	Encode,
+	Decode,
+	DecodeWithMemTracking,
+	RuntimeDebug,
+	TypeInfo,
+	MaxEncodedLen,
+)]
+pub enum PortableForcing {
+	NotForcing,
+	ForceNew,
+	ForceNone,
+	ForceAlways,
+}
+
+// Forcing: RC -> Portable
+impl IntoPortable for pallet_staking::Forcing {
+	type Portable = PortableForcing;
+
+	fn into_portable(self) -> Self::Portable {
+		match self {
+			pallet_staking::Forcing::NotForcing => PortableForcing::NotForcing,
+			pallet_staking::Forcing::ForceNew => PortableForcing::ForceNew,
+			pallet_staking::Forcing::ForceNone => PortableForcing::ForceNone,
+			pallet_staking::Forcing::ForceAlways => PortableForcing::ForceAlways,
+		}
+	}
+}
+
+// Forcing: Portable -> AH
+impl Into<pallet_staking_async::Forcing> for PortableForcing {
+	fn into(self) -> pallet_staking_async::Forcing {
+		match self {
+			PortableForcing::NotForcing => pallet_staking_async::Forcing::NotForcing,
+			PortableForcing::ForceNew => pallet_staking_async::Forcing::ForceNew,
+			PortableForcing::ForceNone => pallet_staking_async::Forcing::ForceNone,
+			PortableForcing::ForceAlways => pallet_staking_async::Forcing::ForceAlways,
+		}
+	}
+}
+
+#[derive(
+	PartialEq,
+	Eq,
+	Clone,
+	Encode,
+	Decode,
+	DecodeWithMemTracking,
+	RuntimeDebug,
+	TypeInfo,
+	MaxEncodedLen,
+)]
+pub struct PortableStakingLedger {
+	pub stash: AccountId32,
+	pub total: u128,
+	pub active: u128,
+	pub unlocking: BoundedVec<PortableUnlockChunk, ConstU32<100>>, /* 100 is an upper bound TODO
```

Comment:
Upper bound is 32 (`Staking::MinUnlockingChunks`)

Reply:
`MaxUnlockingChunks` which is 32 in DOT, KSM and PAS. If you want an upper, 100 is good.